### ISSUE TITLE
Error: Too Many Requests

### ISSUE THREAD

**Issue by Heziode** [vladimiry/email-securely-app#72]
Hi

I get the folowings errors when I use the app: `Too Many Requests`

Logs:
```
[2018-11-22 09:28:01.053] [error] [WEBVIEW:protonmail] [index] [XMLHttpRequest.load handler] {"status":429,"statusText":"Too Many Requests","responseURL":"https://mail.protonmail.com/api/conversations/<token>"}
[2018-11-22 09:28:01.054] [error] [WEBVIEW:protonmail] [index] [XMLHttpRequest.load handler] {"status":429,"statusText":"Too Many Requests","responseURL":"https://mail.protonmail.com/api/conversations/<token>"}
[2018-11-22 09:28:01.055] [error] [WEBVIEW:protonmail] [index] [XMLHttpRequest.load handler] {"status":429,"statusText":"Too Many Requests","responseURL":"https://mail.protonmail.com/api/conversations/<token>"}
[2018-11-22 09:28:01.060] [error] [WEBVIEW:protonmail] [index] [XMLHttpRequest.load handler] {"status":429,"statusText":"Too Many Requests","responseURL":"https://mail.protonmail.com/api/conversations/<token>"}
[2018-11-22 09:28:01.060] [error] [WEBVIEW:protonmail] [index] [XMLHttpRequest.load handler] {"status":429,"statusText":"Too Many Requests","responseURL":"https://mail.protonmail.com/api/conversations/<token>"}
[2018-11-22 09:28:01.060] [error] [WEBVIEW:protonmail] [index] [XMLHttpRequest.load handler] {"status":429,"statusText":"Too Many Requests","responseURL":"https://mail.protonmail.com/api/conversations/<token>"}
[2018-11-22 09:28:01.089] [error] [WEBVIEW:protonmail] [index] [XMLHttpRequest.load handler] {"status":429,"statusText":"Too Many Requests","responseURL":"https://mail.protonmail.com/api/conversations/<token>"}
[2018-11-22 09:28:01.090] [error] [WEBVIEW:protonmail] [index] [XMLHttpRequest.load handler] {"status":429,"statusText":"Too Many Requests","responseURL":"https://mail.protonmail.com/api/conversations/<token>"}
[2018-11-22 09:28:01.090] [error] [WEBVIEW:protonmail] [index] [XMLHttpRequest.load handler] {"status":429,"statusText":"Too Many Requests","responseURL":"https://mail.protonmail.com/api/conversations/<token>"}
[2018-11-22 09:28:01.135] [error] [WEBVIEW:protonmail] [index] [XMLHttpRequest.load handler] {"status":429,"statusText":"Too Many Requests","responseURL":"https://mail.protonmail.com/api/contacts/<token>"}
[2018-11-22 09:28:01.136] [error] [WEBVIEW:protonmail] [index] [XMLHttpRequest.load handler] {"status":429,"statusText":"Too Many Requests","responseURL":"https://mail.protonmail.com/api/contacts/<token>"}
[2018-11-22 09:28:01.136] [error] [WEBVIEW:protonmail] [index] [XMLHttpRequest.load handler] {"status":429,"statusText":"Too Many Requests","responseURL":"https://mail.protonmail.com/api/contacts/<token>"}
```

app verison: 2.0.0-beta-6

electron: 3.0.9
chrome: 66.0.3359.181
node: 10.2.0
v8: 6.6.346.32

**Comment by vladimiry** [comment]
Does it happen when you first time enable the local store feature? The app does an initial data load when the local store is enabled the first time for the account. At the moment app does the fetching without any delays and even more, currently the app runs up to 3 requests of this kind `/api/conversations/<token>` in parallel. It was in general anticipated that Protonmail has some sort of rate limiting enabled at the server side, but it was not triggered on the about 150 conversations case when I was developing the feature, so I guess you have significantly more conversations in your account?

If the guess is correct, then there will be a need to enable delay between requests, which will slow the data load process down but will let it be successfully completed. The thing is that Protonmail doesn't expose its rate limiting settings information and that's understandable. So I think I better disable a parallel loading and enable the configurable delay in `config.json` letting you play around the value.

**Comment by Heziode** [comment]
Yes I have around 500 mails archived.

**Comment by Heziode** [comment]
I have tested with you release.

I have 4 protonmail accounts. I have disabled the local storage settings for each one except for the main wich contains more than 500 mails.
With you default configuration I do not have any problem to synchronize the local storage.

**Comment by vladimiry** [comment]
Thanks for confirming issue resolving, so I keep the default configuration for the final v2 release. The fix is included in just published [v2.0.0-beta.7](https://github.com/vladimiry/email-securely-app/releases/tag/v2.0.0-beta.7) release.
